### PR TITLE
Update Sampling API prior to MTMG integration.

### PR DESCRIPTION
**Pull request by ChuckHastings**
First cut at new sampling function definition to clean up things before MTMG integration.

This overlaps some with @jnke2016 work on heterogeneous sampling... we'll need to resolve a merge order once we've decided on the exact API changes and sequence of work.

Latest thought is that this will be merged into @jnke2016's PR and this will be closed once it's no longer needed.

### CODE REVIEW

File: cpp/include/cugraph/sampling_functions.hpp [hunk 1]
```diff
@@ -240,6 +244,192 @@ biased_neighbor_sample(
...
+          bool store_transposed,
+          bool multi_gpu>
+std::tuple<rmm::device_uvector<vertex_t>,
+           rmm::device_uvector<vertex_t>,
+           std::optional<rmm::device_uvector<weight_t>>,
+           std::optional<rmm::device_uvector<edge_t>>,
+           std::optional<rmm::device_uvector<edge_type_t>>,
+           std::optional<rmm::device_uvector<int32_t>>,
+           std::optional<rmm::device_uvector<size_t>>>
+homogeneous_uniform_neighbor_sample(
```
Comment (jnke2016) [inline cpp/include/cugraph/sampling_functions.hpp:339]: I believe we discussed that the API should have two categories: `Heterogeneous_neighbor_sample` and `Homogeneous_neighbor_sample` where each category supports `uniform` and `bias`

File: cpp/src/sampling/detail/conversion_utilities.cu [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                                     thrust::maximum<label_t>());
+
+  label_map.resize(max_label, handle.get_stream());
+
+  thrust::fill(handle.get_thrust_policy(), label_map.begin(), label_map.end(), label_t{0});
+  thrust::gather(handle.get_thrust_policy(),
+                 std::get<0>(label_to_output_comm_rank).begin(),
+                 std::get<0>(label_to_output_comm_rank).end(),
+                 std::get<1>(label_to_output_comm_rank).begin(),
+                 label_map.begin());
```
Comment (seunghwak) [inline cpp/src/sampling/detail/conversion_utilities.cu:51]: You mean thrust::scatter?

File: cpp/src/sampling/detail/conversion_utilities.cu [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  label_t max_label = thrust::reduce(handle.get_thrust_policy(),
+                                     std::get<0>(label_to_output_comm_rank).begin(),
+                                     std::get<0>(label_to_output_comm_rank).end(),
+                                     label_t{0},
+                                     thrust::maximum<label_t>());
+
+  label_map.resize(max_label, handle.get_stream());
+
+  thrust::fill(handle.get_thrust_policy(), label_map.begin(), label_map.end(), label_t{0});
```
Comment (seunghwak) [inline cpp/src/sampling/detail/conversion_utilities.cu:46]: I guess the fill value should be an int32_t value (not label_t{0}).

**Comment by ChuckHastings** [comment]
Incorporated into @jnke2016 PR... closing this.

**Pull request closed without merging**